core/test/utils.hpp

```diff
@@ -200,7 +228,8 @@ struct TypenameNameGenerator {
 
 
 struct PairTypenameNameGenerator {
-    template <typename T>
+    template <typename T,
+              std::enable_if_t<std::tuple_size<T>::value == 2, bool> = true>
```

Comment:
Nit: That would be a TupleTypenameNameGenerator, not Pair. No need to provide two overloads